src/particle.rs

```diff
@@ -0,0 +1,482 @@
+use std::io::Write;
+
+use anyhow::Context;
+use vek::{Rgb, Vec3};
+
+use crate::block::{BlockPos, BlockState};
+use crate::protocol::{Decode, Encode, VarInt};
+
+#[derive(Clone, Debug)]
+pub struct Particle {
+    pub particle_type: ParticleType,
+    pub long_distance: bool,
+    pub position: Vec3<f64>,
+    pub offset: Vec3<f32>,
+    pub max_speed: f32,
+    pub particle_count: u32,
+}
+
+#[derive(Clone, Debug)]
+pub enum ParticleType {
```

Comment:
I think `Particle` would be a better name for this. To resolve the ambiguity with the packet, the packet can be renamed to  `ParticleS2c`.

Reply:
I agree. Was only hung up on the packet struct being named differently from other packet structs, but I guess it's fine